packages/client-sdk-web/src/internal/data-client.ts

```diff
@@ -1091,9 +1091,9 @@ export class DataClient<
     );
     const result = await this.sendDictionarySetField(
       cacheName,
-      dictionaryName,
-      this.convert(field),
-      this.convert(value),
+      this.convertToB64String(dictionaryName),
```

Comment:
This is the crux of the issue.  You need to b64 the collection name or else it gets stored with a weird bytes key that is not accessible by other SDKs or the CLI.

You need to be careful to do this in *EVERY* collection API.

I think it's also crucial that we add a DELETE test for every collection type.  It won't 100% guarantee that we do the base64 correctly in all APIs but it dramatically increases the odds of us catching a bug.